PsfLauncher/main.cpp

```diff
@@ -462,15 +324,16 @@ ErrorInformation StartWithShellExecute(std::filesystem::path packageRoot, std::f
         , static_cast<WORD>(cmdShow)
     };
 
-
     Log("\tUsing Shell launch: %ls %ls", shex.lpFile, shex.lpParameters);
-    if (!ShellExecuteEx(&shex))
-    {
-        auto err{ ::GetLastError() };
-        return { L"ERROR: Failed to create detoured shell process", err };
-    }
-
-    return GetExitCodeForProcess({ exeName.native().c_str() }, shex.hProcess);
+    THROW_LAST_ERROR_IF_MSG (
+        !ShellExecuteEx(&shex),
+        "ERROR: Failed to create detoured shell process");
+
+    THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_HANDLE), shex.hProcess == INVALID_HANDLE_VALUE);

```

Comment:
>THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_HANDLE), shex.hProcess == INVALID_HANDLE_VALUE); [](start = 4, length = 93)

Instead of throwing this, does it make more sense to throw the static_cast<int>(shex.hInstApp)?

See [https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecuteexa](https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecuteexa)

Reply:
Seems like I can just get last error in this case according to the documentation. I think that is the safest thing to do here.

---
In reply to: [313160279](https://github.com/microsoft/MSIX-PackageSupportFramework/pull/77#discussion_r313160279) [](ancestors = 313160279)